### PR TITLE
Extract theme logic to global store and apply theme to button tooltips

### PR DESCRIPTION
**Pull request by pxlrbt**
- Extract theme logic to a global store. 
- Change button/cell tooltip theme based on global theme

### CODE REVIEW

**Comment by pxlrbt** [comment]
Done.

**Comment by danharrin** [comment]
Do these tooltips work outside of the admin panel now?

**Comment by pxlrbt** [comment]
I don't think so. I only added the theme logic to it. If they didn't work before they probably won't work now.

**Comment by danharrin** [comment]
They did work before, but I wonder how the theming logic that isnt available outside the admin affects that

**Comment by pxlrbt** [comment]
Ah ... damn didn't think about that. I am only using the admin panel.

**Comment by danharrin** [comment]
> Ah ... damn didn't think about that. I am only using the admin panel.

Not expecting you to work further on fixing this if you're not actively benefiting from your work, but I need to mark this as Draft so that I remember to do some work on it myself if you don't want to.

**Comment by pxlrbt** [comment]
@danharrin As I have no experience with the Form and Table Builder as separate packages I'll leave this up to you.

File: packages/tables/resources/views/components/button.blade.php [hunk 1]
```diff
@@ -101,8 +101,8 @@
 @elseif ($tag === 'a')
     <a
         @if ($tooltip)
-            x-data="{}"
-            x-tooltip.raw="{{ $tooltip }}"
+            x-data="tooltip(@js($tooltip))"
+            x-tooltip="tooltip"
```
Comment (wychoong) [inline packages/tables/resources/views/components/button.blade.php:105]: can we just typeof tooltip !== 'undefined'  check
as the concern is on the new `Alpine.data('tooltip')`
and keep `x-tooltip.raw` as fallback for those not using admin panel

`
x-data="typeof tooltip !== 'undefined' ? tooltip(@js($tooltip)) : {}"
x-tooltip.raw="{{ $tooltip }}"
x-tooltip="typeof tooltip !== 'undefined' ? tooltip : null"
`

however i also don't have standalone table project, but i simulated it seems fine

Comment (danharrin) [reply packages/tables/resources/views/components/button.blade.php:105]: So multiple `x-tooltip` attributes on the same element works?

**Comment by zepfietje** [comment]
What's required for this PR to move forward?

**Comment by pxlrbt** [comment]
Currently, this solution only works for the admin panel, but not for the forms package. I don't have any experience with the standalone forms package, therefore I didn't continue work on this. 
I guess the main issue was detecting dark/light mode on the forms package.

**Comment by zepfietje** [comment]
Hmm yeah, so we basically need the theme logic in the packages as well? 🤔

Did you already think of a solution, @danharrin?

**Comment by danharrin** [comment]
Gonna close this, since it's quite far behind `main` and needs to be reworked taking into account recent changes. We need to find a solution for the subpackages as well.

**Pull request closed without merging**